public/resources/js/stats.js

```diff
@@ -915,105 +915,45 @@ document.addEventListener('DOMContentLoaded', function(){
-                            notification.hide();
-                        }, 1500);
-                    }
-                } catch {
-                    let cookieArray = uuid ? [uuid] : [];
-                    setCookie("favorite", JSON.stringify(cookieArray), 365);
+    const favoriteElement = document.querySelector('.favorite');
 
-                    notification.setContent("Set favorite using new system!");
-                    
-                    notification.show();
+    function checkFavorite() {
+        const favorited = parseFavorites(getCookie("favorite")).includes(favoriteElement.getAttribute("data-username"));
+        favoriteElement.setAttribute('aria-checked', favorited);
+        return favorited;
+    }
 
-                    setTimeout(function(){
-                        notification.hide();
-                    }, 1500);
-                }
-            }
-        });
+    let favoriteNotification = tippy(favoriteElement, {
+        trigger: 'manual'
     });
 
-    [].forEach.call(document.querySelectorAll('.remove-favorite'), function (e) {
-        let element = e;
-
-        let notification = tippy(element, {
-            trigger: 'manual'
-        });
-
-        element.addEventListener('click', async function () {
-            let uuid = element.getAttribute("data-username");
-            try {
-                let cookieArray = JSON.parse(getCookie("favorite"));
-                if (cookieArray.includes(uuid)) {
-                    cookieArray.splice(cookieArray.indexOf(uuid), 1);
-
-                    setCookie("favorite", JSON.stringify(cookieArray), 365);
-
-                    notification.setContent("Successfully removed favorite!");
-
-                    notification.show();
-
-                    setTimeout(function () {
-                        notification.hide();
-                    }, 1500);
-                } else {
-                    notification.setContent("That person isn't favorited!");
-                    notification.show();
-
-                    setTimeout(function () {
-                        notification.hide();
-                    }, 1500);
-                }
-            } catch {
-                let cookieArray = uuid ? [uuid] : [];
-                setCookie("favorite", JSON.stringify(cookieArray), 365);
-
-                notification.setContent("Set favorite using new system!");
+    favoriteElement.addEventListener('click', function(){
+        let uuid = favoriteElement.getAttribute("data-username");
+        if(uuid == "0c0b857f415943248f772164bf76795c"){
+            favoriteNotification.setContent("No");
+        }else{
+            let cookieArray = parseFavorites(getCookie("favorite"));
+            if(cookieArray.includes(uuid)){
+                cookieArray.splice(cookieArray.indexOf(uuid), 1);
 
-                notification.show();
+                favoriteNotification.setContent("Removed favorite!");
+            }else{
```

Comment:
I don't see a max amount of favorites here, is it set somewhere else?

Reply:
sigh it didn't select the correct parts whatever

Reply:
thanks, I forgot to add that it should be there now